From 6c43a13cfaa411ee2efc9c9d0747e884b75e1931 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Wed, 11 Sep 2024 17:33:52 +0300
Subject: [PATCH] [FSW-8694] fs1: write on non local address space

This patch add support to write data on other
die address space. For this initiator die has to
disable its address manipulation before actual
data write, enable it back once data write
completes.

Change-Id: I350932482dafad6645760a892852f454fd5168bd
---

diff --git a/include/mem_io.h b/include/mem_io.h
index 763ca3a..1957355 100644
--- a/include/mem_io.h
+++ b/include/mem_io.h
@@ -69,4 +69,17 @@
 	return regmap_readl(ctx->spi_base + offset);
 }
 
+static inline void regmap_writel_inter_die(uint32_t addr_mnpl_ctrl, uint32_t val, uint32_t offset)
+{
+	uint32_t addr_mnpl_ctrl_val;
+
+	addr_mnpl_ctrl_val = regmap_readl(addr_mnpl_ctrl);
+	base_writel(0x0, addr_mnpl_ctrl);
+
+	/* TODO: verify on multi die env and make sure working without barrier */
+	base_writel(val, offset);
+
+	base_writel(addr_mnpl_ctrl_val, addr_mnpl_ctrl);
+}
+
 #endif /* _MEM_IO_H_ */
